src/buffer_manager.cc

```diff
@@ -105,4 +105,29 @@ void BufferManager::clear_buffer_trash()
     m_buffer_trash.clear();
 }
 
+void BufferManager::arrange_buffers(ConstArrayView<String> first_ones)
+{
+    Vector<size_t> indices;
+    for (const auto& name : first_ones)
+    {
+        auto it = find_if(m_buffers, [&](auto& buf) { return buf->name() == name or buf->display_name() == name; });
+        if (it == m_buffers.end())
+            throw runtime_error{format("no such buffer '{}'", name)};
+        size_t index = it - m_buffers.begin();
+        if (contains(indices, index))
+            throw runtime_error{format("buffer '{}' appears more than once", name)};
+        indices.push_back(index);
+    }
+
+    BufferList res;
+    for (size_t index : indices)
+        res.push_back(std::move(m_buffers[index]));
+    for (auto& buf : m_buffers)
```

Comment:
Is it safe to iterate over a `m_buffers` when several items in it have been `std::move`d?

Reply:
Yes, the `if (buf)` check will simply evaluate to false.

Reply:
According to [the documentation](https://en.cppreference.com/w/cpp/utility/move), all `m_buffers[index]` that have been moved "are placed in a valid but unspecified state".

That sounds to me like the `buf` iterator will be valid, but `if (buf)` may or may not be specified (compiler-dependent).

Reply:
You want to look at the documentation of unique_ptr in this case.

Reply:
Where exactly?

Reply:
here: https://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr, the overload #5. 
More specifically: 
>Constructs a unique_ptr by transferring ownership from u to *this and stores the null pointer in u.

Reply:
Makes sense, thanks!